### DISCUSSION TITLE
500 Internal Server Error

### DISCUSSION THREAD

**Discussion by vskmizoram** [Sunbird-cQube/community#859] [Q&A] [answered]
I have again installed the Cqube Ver 5.0.3 and i am facing the issue 500 internal server error .. screenshot attached

![image](https://github.com/Sunbird-cQube/community/assets/157444637/9100d924-8843-41bd-9889-fb21e87d25a1)

at the time of logging in using VSK dashboard

**Comment by vskmizoram** [comment]
@Rajesh .. I hv recreated the dashboard_app ... the same problem is coming

**Reply by bommalla** [reply]
@vskmizoram 

I think in your server some routing issue is there.Need check and get back to you.

**Reply by trigyn-Himanshu** [reply]
@bommalla - VSK Nagaland is facing the same issue, how to resolve this? can you provide more info on routing issue please.

**Reply by PavitraIR** [reply]
@trigyn-Himanshu Please share the screenshot of the issue.

**Comment by bommalla** [comment, marked as answer]
@vskmizoram 

Any update on server routing issue ?

**Comment by vskmizoram** [comment]
Please share the screenshot of the issue

**Reply by trigyn-Himanshu** [reply]
<img width="336" alt="Screenshot 2024-04-22 134616" src="https://github.com/Sunbird-cQube/community/assets/137037616/ff78bc21-b1b4-4127-b728-989ad25524b8">

**Reply by trigyn-Himanshu** [reply]
@vskmizoram  We are experiencing the same issue that you encountered for Mizoram. We need some advice on how to resolve the routing issue, if any, as our cQube installation is fine. I can log in to Keycloak and create users. We are able to see the dashboard but unable to log in.

**Reply by PavitraIR** [reply]
@trigyn-Himanshu Has the routing issue been resolved?  Could you provide details on the cQube version you've installed and specify the type of storage you plan to use?

**Reply by trigyn-Himanshu** [reply]
Hi @PavitraIR  issue still persists, we are using cQube 5.0.3 and storage is local as the VM is hosted in SDC.

**Reply by PavitraIR** [reply]
@trigyn-Himanshu  Could you please provide a screenshot of the error from the network tab? Simply inspect the element and share the screenshot with me.

**Reply by trigyn-Himanshu** [reply]
<img width="940" alt="Screenshot 2024-04-22 174657" src="https://github.com/Sunbird-cQube/community/assets/137037616/f6269b71-f19c-45f4-9539-8135df0ed92e">
PFA

**Reply by PavitraIR** [reply]
@trigyn-Himanshu The issue appears to be related to the SSL certificate please check the server.crt and server.key whether it is matching or not. both cert and key should be match , if yes....Please find the below steps for nginx_ms

cd /opt/cqube/nginx
ls
cat default.conf
Note: Check the all port and Crt and key and domain name
Then copy the new crt and key files in nginx container(from your local to container):

1. sudo docker exec -it niginx_app bash
2. ls
3. cd etc/ssl
4. ls
5. Create one domain directory(example :-- mkdir <domain_name>)
6. cd <domain_name>
7. cd
8. cd ssl_certificates
9. sudo docker cp server.crt nginx_app:/etc/ssl/<domain_name>
10. sudo docker cp server.key nginx_app:/etc/ssl/<domain_name>
11. sudo docker restart nginx_app
12. sudo docker logs nginx_app (Then check the logs of nginx)

**Note: If the login issue persists even after following these steps, it indicates that the SSL certificate has expired.**

**Reply by trigyn-Himanshu** [reply]
@PavitraIR  - Followed the same steps but issue still persists.

root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in# openssl x509 -enddate -noout -in /etc/ssl/vsk.nagaland.gov.in/fullchain.pem
notAfter=Jul 16 15:58:00 2024 GMT
root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in#
root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in#
root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in# ls
cert.pem  chain.pem  fullchain.pem  privkey.pem  vsk.nagaland.gov.in-0001
root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in#
root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in#
root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in#

**Reply by PavitraIR** [reply]
@trigyn-Himanshu If the login issue persists even after following these steps, it indicates that the SSL certificate has expired. Please utilize the new SSL certificate.

Please kindly provide the login page credentials.

**Reply by trigyn-Himanshu** [reply]
@PavitraIR - Generated new SSL certificate, it will expire on 16 july
 root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in# openssl x509 -enddate -noout -in /etc/ssl/vsk.nagaland.gov.in/fullchain.pem
notAfter=Jul 16 15:58:00 2024 GMT
root@5528e82ba402:/etc/ssl/vsk.nagaland.gov.in#

**Reply by PavitraIR** [reply]
@trigyn-Himanshu 
Could you please share the credentials with me if that's possible?

**Reply by PavitraIR** [reply]
@trigyn-Himanshu Thanks, let's verify that and I'll update you accordingly.

**Reply by PavitraIR** [reply]
@trigyn-Himanshu Could you please check the querybuilder logs and keyclock logs ? Here's the command to access the logs. Once you've checked them, please send me a screenshot of the logs.

**1.  sudo docker logs querybuilder_app**



**Note: Have you configured the username and password in Keycloak? Was the configuration successful?**

**Reply by trigyn-Himanshu** [reply]
<img width="777" alt="Screenshot 2024-04-24 120914" src="https://github.com/Sunbird-cQube/community/assets/137037616/90da0683-6448-48d5-b4cc-b7082a48e5ab">

**Reply by trigyn-Himanshu** [reply]
Yes we have created  few users in keycloak and configuration was also successful.

**Reply by trigyn-Himanshu** [reply]
<img width="941" alt="keycloak" src="https://github.com/Sunbird-cQube/community/assets/137037616/509cf6ec-4cb1-4a56-9e85-0f0eacb3e8ce">
Keycloak logs

**Reply by trigyn-Himanshu** [reply]
<img width="581" alt="querybuilder" src="https://github.com/Sunbird-cQube/community/assets/137037616/fb1d0490-149e-408d-aff1-5ae42dc597c0">
Querybuilder logs

**Reply by PavitraIR** [reply]
@trigyn-Himanshu Please review the nginx logs once. If there are no errors in the nginx logs, then the issue might be related to your server's routing.

Because I've tested the login on our instance, and everything is working as expected without any issues. It appears that the problem is with your server routing issue and SSL. Please verify your server and SSL configuration **Follow the steps below,** and the login will work successfully. 

cd /opt/cqube/nginx
ls
cat default.conf
Note: Check the all port and Crt and key and domain name

1. sudo docker exec -it niginx_app bash
2. ls
3. cd etc/ssl
4. ls
5. Create one domain directory(example :-- mkdir <domain_name>)
6. cd <domain_name>
7. cd
8. cd ssl_certificates
9. sudo docker cp server.crt nginx_app:/etc/ssl/<domain_name>
10. sudo docker cp server.key nginx_app:/etc/ssl/<domain_name>
11. sudo docker restart nginx_app
12. sudo docker logs nginx_app (Then check the logs of nginx)


If the everything is correct proceed with the following below steps. First check the certificate key whether bundle is there or not and also check in default.old file whether ports are there or not

1. cd /opt
2. cd cqube
3. ls
4. cd nginix
5. ls
6. cat default.old
7. If everything is okay then go to the (sudo docker exec -it querybuilder_app sh)
8. ls -a
9. cat .env
10. sudo vim .env

**If in keyclock_url domain name is there means like this(url/auth) change it into below url
**http://keycloak_app:8080/auth****

**Reply by trigyn-Himanshu** [reply]
@PavitraIR - it worked! Thank you so much for your guidance.

**Reply by PavitraIR** [reply]
@trigyn-Himanshu  Thanks for the update.

**Comment by vskmizoram** [comment]
It's working now.
